test/one/profiler/test/Assert.java

```diff
@@ -5,37 +5,91 @@
 
 package one.profiler.test;
 
+import java.util.function.BiPredicate;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 public class Assert {
-    private static final Logger log = Logger.getLogger(Assert.class.getName());
+    enum Comparison {
+        GT(">", (a, b) -> a > b),
+        GTE(">=", (a, b) -> a >= b),
+        LT("<", (a, b) -> a < b),
+        LTE("<=", (a, b) -> a <= b),
+        EQ("==", Double::equals),
+        NE("!=", (a, b) -> !Double.isNaN(a) && !Double.isNaN(b) && !a.equals(b));
+
+        public final String operator;
+        public final BiPredicate<Double, Double> comparator;
 
-    public static void isGreater(double value, double threshold) {
-        isGreater(value, threshold, null);
+        Comparison(String operator, BiPredicate<Double, Double> comparator) {
+            this.operator = operator;
+            this.comparator = comparator;
+        }
     }
 
-    public static void isGreater(double value, double threshold, String message) {
-        boolean asserted = value <= threshold;
-        log.log(Level.FINE, "isGreater (asserted: " + asserted + ") " + (message == null ? "" : message) + ": " + value + " > " + threshold);
+    private static final Logger log = Logger.getLogger(Assert.class.getName());
+
+    private static void assertComparison(Comparison comparison, double left, double right, String message) {
+        boolean asserted = !comparison.comparator.test(left, right);
+        String operation = left + " " + comparison.operator + " " + right;
+
+        log.log(Level.FINE, "isAsserted " + asserted + ", "
+                + (message == null ? "" : "'" + message + "'") + ": "
+                + operation);
 
         if (asserted) {
-            throw new AssertionError(
-                    "Expected " + value + " > " + threshold + (message != null ? (": " + message) : ""));
+            String lines = "\n" + SourceLocator.tryGetFrom(new Exception(), 2);
+            String msg = message != null ? (": " + message + "\n" + lines) : lines;
```

Comment:
Construction is a bit hard to read. There'll be an extra empty line if the message is not empty, is it intended?